admin/services/check-window.service.js

```diff
@@ -142,6 +142,84 @@ const checkWindowService = {
     })
 
     return R.concat(newCheckWindows, pastCheckWindows)
+  },
+
+  /**
+   * Format unsaved data
+   * @param {Object} requestData
+   * @returns {Object}
+   */
+  formatUnsavedData: (requestData) => {
+    if (!requestData['adminStartDay'] && !requestData['adminStartMonth'] && !requestData['adminStartYear'] && requestData['existingAdminStartDate'] && requestData['adminIsDisabled']) {
+      requestData.adminStartDay = requestData['existingAdminStartDate'].format('D')
+      requestData.adminStartMonth = requestData['existingAdminStartDate'].format('MM')
+      requestData.adminStartYear = requestData['existingAdminStartDate'].format('YYYY')
+    }
+
+    if (!requestData['checkStartDay'] && !requestData['checkStartMonth'] && !requestData['checkStartYear'] && requestData['existingCheckStartDate'] && requestData['checkStartIsDisabled']) {
+      requestData.checkStartDay = requestData['existingCheckStartDate'].format('D')
+      requestData.checkStartMonth = requestData['existingCheckStartDate'].format('MM')
+      requestData.checkStartYear = requestData['existingCheckStartDate'].format('YYYY')
+    }
+    return requestData
+  },
+
+  /**
+   * Save check window
+   * @param {Object} requestData
+   */
+  save: async(requestData) => {
+    let checkWindow
+    if (requestData.checkWindowId) {
+      checkWindow = await checkWindowDataService.sqlFindOneById(requestData.checkWindowId)
```

Comment:
Should this await be wrapped in a try/catch in case it is rejected?

Reply:
there's a try/catch in the controller level.
It should capture the rejection in case it occurs.